module/zcommon/zfeature_common.c

```diff
@@ -753,6 +753,18 @@ zpool_feature_init(void)
 	    "org.openzfs:raidz_expansion", "raidz_expansion",
 	    "Support for raidz expansion",
 	    ZFEATURE_FLAG_MOS, ZFEATURE_TYPE_BOOLEAN, NULL, sfeatures);
+	{
+		static const spa_feature_t special_failsafe_deps[] = {
+			SPA_FEATURE_ALLOCATION_CLASSES,
+			SPA_FEATURE_NONE
+		};
```

Comment:
IIRC dedup (which you mentioned also) predates allocation classes, so in theory this dependency might be excessive. Though I can't speak about implementation yet.